src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const allProperties = sourceString.split(';');
+
+  const resultObject = allProperties.reduce((currentObject, property) => {
+    const part = property.split(':');
+    const partName = part[0]?.trim();
+    const partValue = part[1]?.trim();
```

Comment:
you can do something like this to simplify your code

```suggestion
    const [partName, partValue] = property.split(':').map(str => str.trim());
```